torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -1358,6 +1358,23 @@ void decomposeLinearOps(Block* block) {
   }
 }
 
+// This is temporary to handle intermediate tensor inserted by autodiff is not
+// being profiled
+void markMissingType(Block* block) {
+  std::vector<Node*> linear_nodes;
+  for (Node* n : block->nodes()) {
+    for (Block* b : n->blocks()) {
+      markMissingType(b);
+    }
+    // fill in the tensor type for mask output in `aten::native_dropout`
+    if (n->kind() == aten::native_dropout) {
```

Comment:
this seems a bit fragile: we're just checking the kind w/o considering the schema - in the unlikely case we get a different another native_dropout overload or the schema changes the hardcoded outputs()[1] and [0] may not be what we expect.

am I overthinking it?

Reply:
Good catch and you are not over thinking this. Let me switch to a schema match just to be on the safer side.